### PR TITLE
Form

### PR DESCRIPTION
**Pull request by ArPimentel**
[x] Created the recipe-form component to handle recipe submissions.
[x] Implemented form validation to ensure all fields are filled in the recipe form.
[x] Refactored multiple components for improved code organization and readability.
[x] Implemented the RecipesService to handle HTTP requests and fetch recipe data.
[x] Added error handling for form submissions and displayed error messages using an error modal.
[x] Developed the SuccessModalComponent to show a success message upon successful recipe addition.
[x] Made CSS adjustments to center elements and enhance visual consistency.
[x] Updated the error-modal component's styles to improve its visual presentation.

### CODE REVIEW

**Review by remy-rm — DISMISSED** [review]
it’s good for me just see with benj if it’s good to change the angular version

File: src/app/recipe-form/recipe-form/recipe-form.component.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        <label for="diet">Diet</label>
+        <div *ngIf="isFieldInvalid('diet')" class="error-message">Diet is required</div>
+        <input type="text" id="diet" name="diet" formControlName="diet" required />
+
+        <label for="steps">Steps</label>
+        <div *ngIf="isFieldInvalid('steps')" class="error-message">Steps is required.</div>
+        <textarea id="steps" name="steps" formControlName="steps" required></textarea>
+
+        <!-- Rest of the form fields -->
```
Comment (Benja-san) [inline src/app/recipe-form/recipe-form/recipe-form.component.html:52]: remove dead code

File: src/app/recipe-form/recipe-form/recipe-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+  onCancel() {
+    this.recipeForm.reset();
+    
+  }
+
+  onFileChange(event: any) {
+    const file = event.target.files[0];
+    if (file) {
+      // Vérifier le type de fichier
```
Comment (Benja-san) [inline src/app/recipe-form/recipe-form/recipe-form.component.ts:97]: remove dead code

File: src/app/recipe-form/recipe-form/recipe-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+      const fileType = file.type;
+      if (fileType === 'image/jpeg' || fileType === 'image/png' || fileType === 'video/mp4') {
+        // Vérifier la taille du fichier
+        const fileSize = file.size;
+        const maxSize = 1024 * 1024 * 1024; // 1 GB
+        if (fileSize <= maxSize) {
+          // Effectuer des opérations avec le fichier
+          console.log('Nom du fichier:', file.name);
```
Comment (Benja-san) [inline src/app/recipe-form/recipe-form/recipe-form.component.ts:105]: remove unnecessaries log

File: src/app/services/recipies/recipes.service.ts [hunk 4]
```diff
@@ -57,6 +60,24 @@ export class RecipesService {
     );
   }
 
+  
+  saveRecipe(recipe: any): Observable<any> {
```
Comment (Benja-san) [inline src/app/services/recipies/recipes.service.ts:64]: avoid any type

File: src/app/services/recipies/recipes.service.ts [hunk 5]
```diff
@@ -57,6 +60,24 @@ export class RecipesService {
...
+    // TODO: Implement actual recipe saving logic here
```
Comment (Benja-san) [inline src/app/services/recipies/recipes.service.ts:65]: keep TODO remove other line

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Pull request closed without merging**